pkg/admission/server.go

```diff
@@ -137,17 +132,24 @@ func ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	review := admission.AdmissionReview{}
-	err = json.Unmarshal(data, &review)
+	res, err := ensureKindAdmissionReview(data)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	if !ensureKindAdmissionReview(data) {
+	if !res {
```

Comment:
It looks like we're superfluously running `json.Unmarshal` _twice_ for any particular data :thinking: 

I'm inclined to say we should just do it once around (what is currently) line 135, and then here just check the kind?

e.g. something like:

```go
	msg, err := marshal(data)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 	if msg.Kind != "AdmissionReview" {
```

Reply:
Make sense, updated.